src/parquet/util/input-output-test.cc

```diff
@@ -50,6 +50,31 @@ TEST(TestInMemoryOutputStream, Basics) {
   ASSERT_TRUE(data_buf.Equals(*buffer));
 }
 
+TEST(TestBufferedReader, Basics) {
+  std::vector<uint8_t> data = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
+  auto buffer = std::make_shared<Buffer>(data.data(), data.size());
+  BufferReader reader(buffer);
+
+  uint8_t out[4];
+  reader.Read(4, out);
```

Comment:
Can you test reading past the end of the buffer and check that the position is incremented appropriately? Perhaps also do this with the other `Read` API, too, and verify that the returned buffer `size()` is as expected